src/frontend/calendar/utils.js

```diff
@@ -41,3 +47,66 @@ export function getRows( year, month ) {
 
 	return rows;
 }
+
+/**
+ * Get human-friendly reccurance string.
+ *
+ * @param {Object} event
+ */
+export function getFrequencyLabel( event ) {
+	const occurrences = {
+		1: __( '1st', 'wporg' ),
+		2: __( '2nd', 'wporg' ),
+		3: __( '3rd', 'wporg' ),
+		4: __( '4th', 'wporg' ),
+	};
+	const dayOfWeek = format( 'l', event.datetime );
+
+	switch ( event.recurring ) {
+		case 'weekly':
+			return sprintf( __( 'Every week on %s', 'wporg' ), dayOfWeek );
+
+		case 'biweekly':
+			return sprintf(
+				__( 'Every other week on %s', 'wporg' ),
```

Comment:
I'm sure it's used elsewhere as well, but I'm just noting that "every other week" is somewhat US-English-specific. Australian and British English would tend to use "every second week", possibly other English locales also. It's definitely better than "biweekly" which is a term I wish would die. Usually this would be a job for translation but I think we'll be running this in a non-localized context. I don't have any better ideas, we can probably just run with it and see if anyone finds it an issue in practice.

Reply:
interesting, I had no idea that was a US-ism.